jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java

```diff
@@ -854,4 +850,35 @@ public void remove()
             };
         }
     }
+
+    /**
+     * Support for Jetty `jetty-loader:` protocol.
+     *
+     * <p>
+     * Internally uses {@link Loader#getResource(String)} to find URL
+     * </p>
+     */
+    public static class JettyLoaderURLStreamHandlerProvider extends URLStreamHandlerProvider
+    {
+        @Override
+        public URLStreamHandler createURLStreamHandler(String protocol)
+        {
+            if ("jetty-loader".equals(protocol))
+            {
+                return new URLStreamHandler()
+                {
+                    @Override
+                    protected URLConnection openConnection(URL u) throws IOException
+                    {
+                        String path = u.getPath();
+                        URL url = Loader.getResource(path);
```

Comment:
This is a scary change of behaviour.  Previously the dtd/schemas were obtained by essentially `XmlConfiguration.class.getResource(String)`.   Now they are being directly loaded from the thread context classloader or the system classloader.

Are we sure that multiple versions of Jetty can no co-exist in the same JVM?    What happens if a future version wants to change the implementation of this URLStreamHandler?

Reply:
The Servlet descriptors, dtds/xsds, were loaded with the `Loader.getResource(String)` before.

This protocol handler `jetty-loader` preserves that.

Reply:
Nope: https://github.com/eclipse/jetty.project/blob/jetty-11.0.11/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java#L1906-L1911

Reply:
Lets take a look at the history of the ee10 `WebDescriptor`.

https://github.com/eclipse/jetty.project/commits/jetty-12.0.x/jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebDescriptor.java

First it looked like this ...

https://github.com/eclipse/jetty.project/blob/04acdb72f041a15267bb79c36d8fc4d517beaadb/jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebDescriptor.java#L109-L114

Then I discovered the missing entries, bad entry names, bad JPMS setups, and it became like this ...

https://github.com/eclipse/jetty.project/blob/e848dd08522b186f94ffe8c37397cc07acb3b3b1/jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebDescriptor.java#L109-L114

You can find `Loader.getRequiredResource(String)` here

https://github.com/eclipse/jetty.project/blob/e848dd08522b186f94ffe8c37397cc07acb3b3b1/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Loader.java#L38-L61

As you can see, from the Servlet point of view, it's been Loader.getResource() the entire time.

The loading of the `jetty-xml` schemes never use the `jetty-loader` URL pattern and is still using `XmlParser.class.getResource(String)` in the current codebase.

https://github.com/eclipse/jetty.project/blob/dd5aec090865c10390398d2829588f55775cb282/jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java#L94

and

https://github.com/eclipse/jetty.project/blob/dd5aec090865c10390398d2829588f55775cb282/jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java#L1999